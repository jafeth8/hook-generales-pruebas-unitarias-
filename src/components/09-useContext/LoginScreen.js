import React, { useContext } from 'react'
import { UserContext } from './UserContex'

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext)
    const addUser=()=> {
        setUser({user:1234,name:'el prro'})
    }
    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>
            
            <button className="btn btn-primary" onClick={addUser}>
                nuevo usuario
            </button>
        </div>
    )
}
