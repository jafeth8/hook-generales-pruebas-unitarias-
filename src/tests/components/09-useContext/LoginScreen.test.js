import {mount} from 'enzyme'
import { LoginScreen } from "../../../components/09-useContext/LoginScreen"
import { UserContext } from '../../../components/09-useContext/UserContex'

describe('pruebas en el componente LoginScreen', () => {
    
    const setUser=jest.fn();
  
    const wrapper =mount(
        <UserContext.Provider value={{ setUser}}>
            <LoginScreen/>
        </UserContext.Provider>
    )

    

    test('debe de mostrar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de ejecutar el setUser con el argumento esperado', () => {
        
        const button=wrapper.find('button');
        //console.log(button.props());
        const eventoClik= button.prop('onClick');
        eventoClik()
        expect(setUser).toHaveBeenCalledWith({user:1234,name:'el prro'});
    })
    

})
