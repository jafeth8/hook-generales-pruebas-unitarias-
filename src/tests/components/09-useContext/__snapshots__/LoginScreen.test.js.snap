// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente LoginScreen debe de mostrar correctamente 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    nuevo usuario
  </button>
</div>
`;
